src/java.base/share/classes/java/lang/reflect/MemberPattern.java

```diff
@@ -410,9 +412,9 @@ public Object[] invoke(Object matchCandidate)
                             .map(PatternBinding::getType)
                             .toArray(Class[]::new)
             );
-            MethodHandle initializingConstructor = Carriers.initializingConstructor(bindingMT);
+            MethodHandle initializingConstructor = SharedSecrets.getJavaLangRuntimeAccess().initializingConstructor(bindingMT);
 
-            return (Object[])Carriers.boxedComponentValueArray(bindingMT).invoke(method.invoke(matchCandidate, matchCandidate, initializingConstructor));
+            return (Object[])SharedSecrets.getJavaLangRuntimeAccess().boxedComponentValueArray(bindingMT).invoke(method.invoke(matchCandidate, matchCandidate, initializingConstructor));
```

Comment:
Would `invokeExact` work here?

Reply:
I realized I changed this mechanically, but reflection does not need to go through the Carriers at all. Fixed in [47dd83d](https://github.com/biboudis/amber/pull/30/commits/47dd83de45f9e1ba6fc9803ae89ce67f5f27fe86)